corehq/apps/sms/management/commands/run_sms_queue.py

```diff
@@ -53,11 +66,27 @@ def enqueue(self, queued_sms):
     def handle(self, **options):
         while True:
             try:
-                self.create_tasks()
+                if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS and not can_send_sms_now():
+                    sleep(self._time_till_next_window())
+                else:
+                    self.create_tasks()
             except:
                 notify_exception(None, message="Could not fetch due survey actions")
             sleep(10)
 
+    def _time_till_next_window(self):
+        # if from time is greater than the start_time, it would give seconds till next day start
+        server_now_time = self._get_server_time()
+        today = datetime.date.today()
+        from_datetime = datetime.datetime.combine(today, server_now_time)
+        to_datetime = datetime.datetime.combine(today, datetime.time(9, 0))
+        return (to_datetime - from_datetime).seconds
+
+    def _get_server_time(self):
```

Comment:
server time should always be UTC right?

Reply:
I will name that better to `_get_local_time`.

Reply:
similar to below - I don't see why you need to do all the timezone conversions here. Can't you just compare the UTC times?

Reply:
Hey yes, i did already think of improving that when i actually implement it.